packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -2412,7 +2418,7 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   /// Paste text from [Clipboard].
   @override
-  Future<void> pasteText(SelectionChangedCause cause) async {
+  Future<void> pasteText(SelectionChangedCause cause, {String? text}) async {
```

Comment:
Maybe instead of adding an optional API we can extract most of this paste logic into a private method? Then this method can call that private method with its clipboard text, and the text processing action can also call that private method with its processed text. What do you think?

Reply:
Good idea :+1: . Because `pasteText` is public it was a bad move to introduce this optional parameter.
I tried the private method approach you suggested in the last commit where I introduced `_pasteText` and `_allowPaste`.